Client/Ecosia/Braze/BrazeService.swift

```diff
@@ -0,0 +1,160 @@
+
+    @MainActor
+    private func initBraze(userId: String) throws {
+        self.braze = Braze(configuration: try getBrazeConfiguration())
+        let inAppMessageUI = BrazeInAppMessageUI()
+        inAppMessageUI.delegate = self
+        self.braze?.inAppMessagePresenter = inAppMessageUI
+        Task.detached(priority: .medium) { [weak self] in
+            await self?.updateID(self?.userId)
+        }
+    }
+}
+
+extension BrazeService {
+    // MARK: - Notification Center
+
+    private func makeNotificationCenter(notificationCenterDelegate: UNUserNotificationCenterDelegate) -> UNUserNotificationCenter {
+        let center = UNUserNotificationCenter.current()
+        center.setNotificationCategories(BrazeKit.Braze.Notifications.categories)
+        center.delegate = notificationCenterDelegate
+        return center
+    }
+
+    private func retrieveUserCurrentNotificationAuthStatus() async {
+        let notificationCenter = UNUserNotificationCenter.current()
+        let currentStatus = await notificationCenter.notificationSettings().authorizationStatus
+        notificationAuthorizationStatus = currentStatus
+    }
+}
+
+extension BrazeService {
+    // MARK: - ID Update
+
+    private func updateID(_ id: String?) async {
+        guard let id else { return }
+        #if MOZ_CHANNEL_FENNEC
+        print("📣🆔 Braze Identifier Updating To: \(id)")
+        #endif
+        let brazeID = await braze?.user.id()
+        guard id != brazeID else { return }
+        braze?.changeUser(userId: id)
+    }
+}
+
+extension BrazeService {
+    // MARK: - Environment Configuration
+
+    /// Retrieves the Braze configuration based on the provided parameters.
+    ///
+    /// - Parameters:
+    ///   - apiKey: The Braze API key to be used for configuration.
+    ///   - environment: The target environment for which the Braze configuration is requested.
+    ///                  Defaults to the current environment.
+    /// - Returns: A Braze configuration if the required parameters are present.
+    /// - Throws: An `Error.invalidConfiguration` if the API key is empty.
+    ///
+    /// - Note: The `environment` parameter allows customization of the target environment.
+    ///   If not provided, it defaults to the current environment.
+    ///
+    /// - Warning: Ensure that the provided API key is not empty to avoid invalid configurations.
+    func getBrazeConfiguration(apiKey: String = BrazeService.apiKey,
+                               environment: Environment = Environment.current) throws -> BrazeKit.Braze.Configuration {
+        guard !apiKey.isEmpty else { throw Error.invalidConfiguration }
+
+        let brazeConfiguration = BrazeKit.Braze.Configuration(apiKey: apiKey, endpoint: environment.urlProvider.brazeEndpoint)
+        #if MOZ_CHANNEL_FENNEC
+        brazeConfiguration.logger.level = .debug
+        #endif
+        return brazeConfiguration
+    }
+}
+
+extension BrazeService: BrazeInAppMessageUIDelegate {
```

Comment:
I understand that making it generic would perhaps include the need to mirror all these braze-specific events. It seems that they could be easily managed in the Braze concrete implementation (like the one we have here) in case `Braze` is kept as `EngagementService`'s provider.

Reply:
Yeah, I guess that's the one advantage linked to [the other comment above](https://github.com/ecosia/ios-browser/pull/812#discussion_r1843538630), those private extension methods would be better split into a Braze specific class (even if this already is that) 🤔